### ISSUE TITLE
Stream endpoint

### ISSUE THREAD

**Issue by rullzer** [nextcloud/server#9142]
Assume:

1. A mobile user U using the Nextcloud App N that wants to view a video file in an external app
2. A video app like VLC that you trust
3. A mallware app EVIL that tries to hijack the url send from N to VLC

Because we are sending the link into the open basically we must assure:

1. The link is read only
2. The link is only valid for a limited amount of time (8 hours?)
3. The link does not contain the users username and password
  * Just a random generated token

Basically we'd need an endpoint where the app can say: "I want to have a link to read the file with fileID X". The endpoint will then generate a link that can be send to VLC.

This endpoint should be webdav since it already support range requests by default etc.

CC: @tobiasKaminsky

**Comment by tobiasKaminsky** [comment]
:+1: this is a long wished feature, great that you find a good way to tackle this :tada:
